src/System.Private.CoreLib/src/Internal/Runtime/CompilerServices/OpenMethodResolver.cs

```diff
@@ -174,6 +174,16 @@ unsafe public static IntPtr ResolveMethod(IntPtr resolver, object thisObject)
             return TypeLoaderExports.OpenInstanceMethodLookup(resolver, thisObject);
         }
 
+        unsafe public static IntPtr ResolveMethod(IntPtr resolverPtr, RuntimeTypeHandle thisType)
+        {
+            OpenMethodResolver* resolver = ((OpenMethodResolver*)resolverPtr);
+            IntPtr nonVirtualOpenInvokeCodePointer = resolver->_nonVirtualOpenInvokeCodePointer;
+            if (nonVirtualOpenInvokeCodePointer != IntPtr.Zero)
+                return nonVirtualOpenInvokeCodePointer;
+
+            return RuntimeImports.RhResolveDispatchOnType(thisType.ToEETypePtr(), resolver->_declaringType, checked((ushort)resolver->_methodHandleOrSlotOrCodePointer.ToInt32()));
```

Comment:
```suggestion
            return RuntimeImports.RhResolveDispatchOnType(thisType.ToEETypePtr(), resolver->_declaringType, (ushort)resolver->_methodHandleOrSlotOrCodePointer);
```

You can also fix the other place that converts `_methodHandleOrSlotOrCodePointer` back to slot to a direct cast to `ushort` without `ToInt32`.